calyx-py/test/correctness/queues/binheap/pifo.py

```diff
@@ -0,0 +1,131 @@
+# pylint: disable=import-error
+import sys
+import calyx.builder as cb
+import calyx.queue_call as qc
+from stable_binheap import insert_stable_binheap
+
+
+def insert_flow_inference(comp, value, flow, boundary, group):
+    """The flow is needed when the command is a push.
+    If the value to be pushed is less than or equal to `boundary`,
+    the value belongs to flow 0.
+    Otherwise, the value belongs to flow 1.
+
+    This method adds a group to the component `comp` that does this.
+    1. Within component `comp`, creates a group called `group`.
+    2. Within `group`, creates a cell `cell` that checks for less-than.
+    3. Puts the values `boundary` and `value` into the left and right ports of `cell`.
+    4. Then puts the answer of the computation into `flow`.
+    5. Returns the group that does this.
+    """
+    cell = comp.lt(32)
+    with comp.group(group) as infer_flow_grp:
+        cell.left = boundary
+        cell.right = value
+        flow.write_en = 1
+        flow.in_ = cell.out
+        infer_flow_grp.done = flow.done
+    return infer_flow_grp
+
+
+def insert_binheap_pifo(prog, name, boundary, factor):
+    """Inserts the component `pifo` into the program.
+
+    It is a two-flow, round-robin queue implemented via binary heap
+
+    It has:
+    - two inputs, `cmd` and `value`.
+    - one memory, `mem`, of size `2**queue_size_factor`.
+    - two ref registers, `ans` and `err`.
+    """
```

Comment:
Here too, you'll want a little more commentary to explain the strategy.

Reply:
Attempted in [03e0a00](https://github.com/calyxir/calyx/pull/2164/commits/03e0a0035b8070c3471e6911cec8ecff640eab7b). 

Do you think this includes sufficient detail?